suite/tests/api/ir_riscv64.h

```diff
@@ -0,0 +1,100 @@
+/* **********************************************************
+ * Copyright (c) 2024 Institute of Software Chinese Academy of Sciences (ISCAS).
+ * All rights reserved.
+ * **********************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of ISCAS nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL ISCAS OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+static byte buf[8192];
+
+#ifdef STANDALONE_DECODER
+#    define ASSERT(x)                                                                 \
+        ((void)((!(x)) ? (fprintf(stderr, "ASSERT FAILURE (standalone): %s:%d: %s\n", \
+                                  __FILE__, __LINE__, #x),                            \
+                          abort(), 0)                                                 \
+                       : 0))
+#else
+#    define ASSERT(x)                                                                \
+        ((void)((!(x)) ? (dr_fprintf(STDERR, "ASSERT FAILURE (client): %s:%d: %s\n", \
+                                     __FILE__, __LINE__, #x),                        \
+                          dr_abort(), 0)                                             \
+                       : 0))
+#endif
+
+static byte *
```

Comment:
Always mark functions in header files as `inline` to avoid multiple defs.
(An alternative would be a new .c file containing these.)

Update: the buf[] above needs to be a in .c so maybe this should be also.

Reply:
> Always mark functions in header files as `inline` to avoid multiple defs.

Is `static` itself not enough? Defined as static, every .c includes this header will have a private (static) copy of this function, no duplicated symbol could occur during linking.

> (An alternative would be a new .c file containing these.)
> 
> Update: the buf[] above needs to be a in .c so maybe this should be also.